desc/objectives/_coils.py

```diff
@@ -1348,6 +1348,15 @@ class ToroidalFlux(_Objective):
         zeta=jnp.array(0.0), NFP=eq.NFP).
     name : str, optional
         Name of the objective function.
+    use_vector_potential : True
```

Comment:
could we instead just do 
```python
try:
    A = field.compute_vector_potential(...)
    Phi = dot(A, e_theta)
except ValueError:
    B = field.compute_magnetic_field(...)
    Phi = dot(B, e_zeta)
```

Reply:
yes but it affects which data needs to be computed, if we are fine with extra data being computed I can make it this way